src/QuadForm/ClosestVector.jl

```diff
@@ -317,25 +317,34 @@ function closest_vectors(G::MatrixElem, L::MatrixElem, c::RingElement)
             QTT1 = R1[1] 
             E = R1[2] 
         end
-        #------------------------------------------------
-        for k in E
-            if (transpose(matrix(QQ,size(k,1),1,k))*Q*matrix(QQ,size(k,1),1,k))[1] + (2*transpose(matrix(QQ,size(k,1),1,k))*L)[1] + c <= 0
-                push!(EE, k)
+        if bool == false
+            for k in E
+                if (transpose(matrix(QQ,size(k,1),1,k))*Q*matrix(QQ,size(k,1),1,k))[1] + (2*transpose(matrix(QQ,size(k,1),1,k))*L)[1] + c <= 0
+                    push!(EE, k)
+                end
+            end
+            return sort!(EE)
+        else
+            for k in E
+                if (transpose(matrix(QQ,size(k,1),1,k))*Q*matrix(QQ,size(k,1),1,k))[1] + (2*transpose(matrix(QQ,size(k,1),1,k))*L)[1] + c == 0
+                    push!(EE, k)
+                end
             end
+            return sort!(EE)
         end
-        return EE
     end
 end
 
 
 @doc Markdown.doc"""
-    closest_vectors(L:ZLat, v:Vector, c::fmpq) -> Vector{Tuple{Vector{Int}, fmpq}} -> Array{Array{fmpz, 1}, 1}
+    closest_vectors(L:ZLat, v:Vector, c::fmpq; bool=false) -> Vector{Tuple{Vector{Int}, fmpq}} -> Array{Array{fmpz, 1}, 1}
   
   
   Returns all vectors `x` in `L` such that `b(v-x,v-x) <= c`, where `b` is the bilinear form on `L`.
-  written by: Simon Brandhorst
+  If the optional argument bool=true then it returns all vectors `x` in `L` such that `b(v-x,v-x) = c`.
+  By default bool=false. This version of the function is contributed by: Simon Brandhorst
```

Comment:
Please remove my name from the docstring. It is unusual to give credit in the docstring.